pkg/v1/layout/write.go

```diff
@@ -221,34 +224,92 @@ func (l Path) WriteFile(name string, data []byte, perm os.FileMode) error {
 // WriteBlob copies a file to the blobs/ directory in the Path from the given ReadCloser at
 // blobs/{hash.Algorithm}/{hash.Hex}.
 func (l Path) WriteBlob(hash v1.Hash, r io.ReadCloser) error {
+	return l.writeBlob(hash, -1, r, nil)
+}
+
+func (l Path) writeBlob(hash v1.Hash, size int64, r io.Reader, renamer func() (v1.Hash, error)) error {
+	if hash.Hex == "" && renamer == nil {
+		// Should we panic, since this is a programming error?
+		return errors.New("writeBlob called an invalid hash and no renamer")
```

Comment:
The stdlib often takes the panic approach in cases like these, but not many libraries take the same approach. What's the style of this repo, @jonjohnsonjr?

Reply:
Panic here seems fine.